src/Data/PreciseDateTime.purs

```diff
@@ -42,6 +46,22 @@ instance boundedPreciseDateTime :: Bounded PreciseDateTime where
 instance showPreciseDateTime :: Show PreciseDateTime where
   show (PreciseDateTime dateTime ns) = "PreciseDateTime (" <> show dateTime <> ") " <> show ns
 
+
+mkPreciseDateTime
+  :: Date
+  -> Hour
+  -> Minute
+  -> Second
+  -> Nanosecond
+  -> PreciseDateTime
+mkPreciseDateTime d h m s ns = unsafePartial fromJust pdt -- FIXME: unsafePartial fromJust
+  where
+  t = Time h m s bottom
+  dt = DateTime d t
+  adjustment = Nanoseconds (BigInt.fromInt $ fromEnum ns)
+  pdt = adjust adjustment $ PreciseDateTime dt bottom
```

Comment:
The idea with using `adjust` here was to keep rounding logic in a single place. The nanoseconds might round up to a whole millisecond, which might round up to a whole second, and so on.

This doesn't currently work because `adjust` uses `truncated`.

It also creates the problem of dealing with a `Maybe`. Perhaps it's not unsafe to use `fromJust` because all of the date components are bounded, but I haven't put much thought into it yet.